### PR TITLE
Update NVIDIA ofa_kernel SPEC

### PR DESCRIPTION
**Pull request by jcamposeco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Current OFA kernel Spec is not building with the latest kernel, so it is hardcoded to use 5.15.87 to succeed, the change move this to be able to build with latest kernel.

###### Change Log  

- Remove the condition for kernel 5.15.87
- Update OFA-kernel version package
- Remove obsoletes not used

###### Does this affect the toolchain?  

NO

###### Associated issues  

- #xxxx

###### Test Methodology

- Pipeline build id: 
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=386710&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=386711&view=results
- RUN_CHECK: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=387316&view=results

### CODE REVIEW

**Review by rlmenge — CHANGES_REQUESTED** [review]
Please change the target branch from `2.0` to `main`

**Review by rlmenge — COMMENTED** [review]
Please confirm this package builds on x86 as it does not specifiy `ExclusiveArch`

**Comment by jcamposeco** [comment]
> Please confirm this package also builds on x86 as it does not specifiy `ExclusiveArch`

Build for both X86 and ARM64
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=386710&view=results 
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=386711&view=results

File: SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec [hunk 1]
```diff
@@ -413,75 +372,73 @@ symlink="%{_prefix}/src/ofa_kernel/default"
...
 %{_prefix}/src/ofa_kernel/%{_arch}/%{KVERSION}
 
 %files source
 %defattr(-,root,root,-)
-%{_prefix}/src/ofa_kernel-%version/source
-%{_prefix}/src/mlnx-ofa_kernel-%version
+%{_prefix}/src/ofa_kernel-%{version}/source
+%{_prefix}/src/mlnx-ofa_kernel-%{version}
 
 %changelog
```
Comment (anphel31) [inline SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec:443]: please add an entry to the changelog

File: SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec [hunk 2]
```diff
@@ -25,133 +25,91 @@
 # and/or other materials provided with the distribution.
 #
 #
-
+Summary:        Infiniband HCA Driver
+Name:           mlnx-ofa_kernel
+Version:        23.04
+Release:        3%{?dist}
```
Comment (anphel31) [inline SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec:31]: since you've bumped the version, you can reset the release number back to 1

**Review by anphel31 — APPROVED** [review]

File: SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec (hunk 1 above)
Comment (jcamposeco) [reply SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec:443]: Added

File: SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec (hunk 2 above)
Comment (jcamposeco) [reply SPECS/mlnx-ofa_kernel/mlnx-ofa_kernel.spec]: Done

**Comment by jcamposeco** [comment]
> Please change the target branch from `2.0` to `main`

@rlmenge Changes requested are done, could you please give your heads up?

**Review by rlmenge — APPROVED** [review]

**Pull request merged**